### DISCUSSION TITLE
I'd like a way to export to yaml without requiring maintaining a completely different stack

### DISCUSSION THREAD

**Discussion by sheldonhull** [pulumi/pulumi#9366] [Pulumi Providers]
I've read a lot on this in Slack and GitHub issues.
I'm still not finding what I need. Most have to do with multiplexing the writing to yaml, or such.

For development, I find the yaml a great way to proof what I'm doing in Kuberentes.
However, if I have a complex configuration with secret values, I have to maintain a parallel stack called "yamlrender" to get this to work. 

What I really want to do is figure out a way to run the plan with local state and not have to maintain a duplicate stack configuration just for YAML = true.

Is there anyway that would allow me to iterate more easily with local yaml being my target until I'm ready to actually deploy and not require duplicate stack maintenance?

**Comment by cnunciato** [comment]
@sheldonhull I'm not too familiar with what you're trying to do (don't do much with Kubernetes), so apologies in advance for that -- but are you saying you want to use the `renderYamlToDirectory` property conditionally? Or something else? Are you using the "yamlrender" stack just to configure that value?
